### ISSUE TITLE
Allow texelFetch(), etc. with just a texture2D

### ISSUE THREAD

**Issue by tangent-vector** [KhronosGroup/glslang#1003]
Issue #320 saw a lot of discussion on this topic, and ultimately came to the conclusion that the (non-normative) `KHR_vulkan_glsl` requires `texelFetch()` and similar operations should take `sampler*` arguments for consistency with their existing GLSL definitions.

However, SPIR-V does not reflect this choice, so that the source language semantics require a sampler that is then ignored in the SPIR-V. HLSL does not require a sampler for its equivalent operations (e.g., `Texture2D.Load()`) and can be used to generate SPIR-V without a useless sampler parameter declaration.

Just having a useless sampler *declared* isn't so bad,  it when using Vulkan validation layers, an application ultimately needs to allocate space for that sampler in descriptor tables and pipeline layouts, and has to bind a valid sampler object at runtime, which will then be ignored by the shader code.

Long story short: I'd like to request that glslang support `texelFetch` and similar operations with just a `texture*` as additional overloads, that map directly to SPIR-V without a dummy sampler. If this needs to written up as a (non-normative) extension spec, I can try to do that.

The context here is a tool that generates GLSL source from an HLSL-like language (but which cant use glslang's HLSL support directly), and which therefore doesn't have a sampler on hand to use in these operations.

**Comment by johnkslang** [comment]
Yes, a change to GLSL semantics needs to be triggered by an extension.  Otherwise, I don't see any problem here.

**Comment by aejsmith** [comment]
Any update on this? As already detailed it's pretty irritating to have to bind a dummy sampler for this that doesn't even get used (the SPIR-V code generated combines the image and sampler, and then just extracts the image again for OpImageFetch).

As far as I can see a sampler is not required for texelFetch(Offset), as well as textureSize, textureSamples and textureQueryLevels.

**Comment by sheredom** [comment]
I'll get this discussed in the working group and get back to you asap.

**Comment by johnkslang** [comment]
If the conclusion here is confined to new GLSL semantics, note we now have https://github.com/KhronosGroup/GLSL for doing things like that, and the GLSL part should be defined there.

**Comment by sheredom** [comment]
Yeah I've been talking with @aejsmith offline and I think this should be entirely containable in a public extension + corresponding glslang change, will see what we can do.

**Comment by aejsmith** [comment]
I've sent pull requests for an extension spec (https://github.com/KhronosGroup/GLSL/pull/18) and an implementation of that in glslang (https://github.com/KhronosGroup/glslang/pull/1416)

**Comment by aejsmith** [comment]
Both pull requests have been merged, I think this can be closed now?
